blobber/amazons3_backend.py

```diff
@@ -55,6 +55,4 @@ def upload_to_AmazonS3(hashalgo, blobhash, data_file, headers, metadata):
             key.set_acl('public-read')
 
     # return the blob URL
-    amazon_baseurl = "http://" + BUCKET + ".s3.amazonaws.com"
-    blob_url = urlparse.urljoin(amazon_baseurl, _key)
-    return blob_url
+    return key.generate_url(expires_in=0)
```

Comment:
The last line should be `return key.generate_url(expires_in=0, query_auth=False)`, otherwise boto returns a signed URL, which is expired.